docs/integrations/sources/s3.md

```diff
@@ -120,7 +120,7 @@ As you can probably tell, there are many ways to achieve the same goal with path
 
 ## User Schema
 
-Providing a schema allows for more control over the output of this stream. Without a provided schema, columns and datatypes will be inferred from each file and a superset schema created. This will probably be fine in most cases but there may be situations you want to enforce a schema instead, e.g.:
+Providing a schema allows for more control over the output of this stream. Without a provided schema, columns and datatypes will be inferred from the first file in the bucket matching your path pattern and suffix. This will probably be fine in most cases but there may be situations you want to enforce a schema instead, e.g.:
```

Comment:
I think "first" here can lead to confusion. It's actually the "first created file" / "oldest' as we sort objects by creation date.